src/sentry/integrations/msteams/utils.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import absolute_import
+
+from sentry.models import Integration
+from sentry.utils.compat import filter
+from .client import MsTeamsClient
+
+
+def channel_filter(channel, name):
+    # the general channel has no name in the list
+    # retrieved from the REST API call
+    if channel.get("name"):
+        return name == channel.get("name")
+    else:
+        return name == "General"
+
+
+def get_channel_id(organization, integration_id, name):
+    try:
+        integration = Integration.objects.get(
+            provider="msteams", organizations=organization, id=integration_id
+        )
+    except Integration.DoesNotExist:
+        return None
+
+    team_id = integration.external_id
+    client = MsTeamsClient(integration)
+
+    # handle searching for channels first
+    channel_list = client.get_channel_list(team_id)
+    filtered_channels = list(filter(lambda x: channel_filter(x, name), channel_list))
+    if len(filtered_channels) > 0:
+        return filtered_channels[0].get("id")
+
+    # handle searching for users
+    members = client.get_member_list(team_id, None)
+    while True:
```

Comment:
@KKaempen Just to be safe, I think we should use a for loop here that limits the number of passes we will make. If Microsoft introduces a bug where `continuationToken` is there even when it shouldn't be, I don't want this to go into an infinite loop.

Reply:
Are you suggesting we just hardcode a maximum number of passes, or is there a cleverer way of doing it? I guess a non-hardcoded way would be to check if the member array is equal to the last member array, but that seems awfully clunky.

Reply:
@KKaempen Just hardcode the number of passes to what seems like a reasonable maximum.